qiskit_optimization/applications/bin_packing.py

```diff
@@ -31,14 +31,19 @@ class BinPacking(OptimizationApplication):
         https://en.wikipedia.org/wiki/Bin_packing_problem
     """
 
-    def __init__(self, weights: List[int], max_weight: int) -> None:
+    def __init__(
+        self, weights: List[int], max_weight: int, max_number_of_bins: Optional[int] = None
+    ) -> None:
         """
         Args:
             weights: A list of the weights of items
             max_weight: The maximum bin weight capacity
+            max_number_of_bins: The maximum number of bins
```

Comment:
Can we add here that by default this is set to the number of weights to make it clear in the documentation.